src/aui/framemanager.cpp

```diff
@@ -1807,32 +1785,45 @@ void wxAuiManager::LayoutAddPane(wxSizer* cont,
         part.type = wxAuiDockUIPart::typeCaption;
         part.dock = &dock;
         part.pane = &pane;
-        part.button = NULL;
+        part.button = 0;
         part.orientation = orientation;
         part.cont_sizer = vert_pane_sizer;
         part.sizer_item = sizer_item;
         int caption_part_idx = uiparts.GetCount();
         uiparts.Add(part);
 
         // add pane buttons to the caption
-        int i, button_count;
-        for (i = 0, button_count = pane.buttons.GetCount();
-             i < button_count; ++i)
+        int button_count = 0;
+        const int N = 3;
+        wxAuiButtonId buttons[N] = {
+            wxAUI_BUTTON_MAXIMIZE_RESTORE,
+            wxAUI_BUTTON_PIN,
+            wxAUI_BUTTON_CLOSE
+        };
+        int flags[N] = {
+            wxAuiPaneInfo::buttonMaximize | wxAuiPaneInfo::buttonMinimize,
```

Comment:
It looks like this is a change in behaviour: currently, the `MAXIMIZE_RESTORE` button seems to be only used if `HasMaximizeButton()` returns true, i.e. if `buttonMaximize` flag is set. Are you sure we also want to use it if `buttonMinimize` is?

Reply:
No, it's my mistake - fixed now.